app/org/sagebionetworks/bridge/models/accounts/UserSessionInfo.java

```diff
@@ -41,6 +42,7 @@ public UserSessionInfo(UserSession session) {
         this.sharingScope = session.getUser().getSharingScope();
         this.username = session.getUser().getUsername();
         this.roles = session.getUser().getRoles();
+        this.dataGroups = session.getUser().getDataGroups();
```

Comment:
Added immutable copies here during construction. Added helper method to do this in a null safe way.